### PR TITLE
fix: do not use `uv` to setup python on windows when conditions are not met

### PR DESCRIPTION
**Pull request by mayeut**
`uv` was still used in `setup_python` on Windows even when `use_uv` was `False`.
This PR fixes `setup_python` to fallback to `build` frontend name instead of `build[uv]` in that case.
Added `can_use_uv` function to make sure wether or not `uv` is usable in a single place rather than duplicate the checks (and potentially forgetting one out of two if the conditions were to change).

### CODE REVIEW

File: cibuildwheel/macos.py [hunk 1]
```diff
@@ -193,20 +193,23 @@ def install_pypy(tmp: Path, url: str) -> Path:
     return installation_path / "bin" / "pypy3"
 
 
+def can_use_uv(python_configuration: PythonConfiguration) -> bool:
+    conditions = (Version(python_configuration.version) >= Version("3.8"),)
+    return all(conditions)
```
Comment (Czaki) [inline cibuildwheel/macos.py:198]: Why not just 
```
return Version(python_configuration.version) >= Version("3.8")
```

**Review by Czaki — APPROVED** [review]
Does the Linux not need similar refactor?

File: cibuildwheel/macos.py (hunk 1 above)
Comment (mayeut) [reply cibuildwheel/macos.py:198]: Because I used the same layout as on Windows for consistency where using a conditions tuple is more readable.

**Comment by mayeut** [comment]
> Does the Linux not need similar refactor?

Linux could if we want to push consistency even further but given it's used in only one place on Linux, the risk of modifying one place and forgetting the other does not exist there.

File: cibuildwheel/macos.py (hunk 1 above)
Comment (Czaki) [reply cibuildwheel/macos.py:198]: ok.

**Review by joerick — APPROVED** [review]

**Pull request merged**